### PR TITLE
Revise testing strategy and update python api

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR adds a test for the python api for create/recreate.

My goal here is to simplify the tests so that they are easier to debug.

- The system is no longer dependent on the global config and instead carries the config as an attribute. I removed direct imports of the global config `CFG` everywhere except in the cli/config/operations modules. 
- Adds api for recreate
- Adds api for submit
- Adds api for status
- Revises create/submit/status tests (to be completed at a later stage)

Needs containerized runs to be available under the `CONTAINERIZED_RUNS_DIR` environment variable.

```
cd $JRUNS
git clone git@github.com:duqtools/containerized_runs
export CONTAINERIZED_RUNS_DIR="$JRUNS/containerized_runs"
```

Addresses #609 

## TODO

- [x] Make output for create/recreate consistant (list of jobs/runs vs single job/run)
- [x] Document recreate
- [x] Document development setup
- [x] Remove v210921 tests (too much work to maintain)
- [x] Fix ETS6 system

### CODE REVIEW

**Comment by stefsmeets** [comment]
Hi @v1kko do you want to have a look at this one? I revised your ETS6 work a bit to factor out the config, can you confirm that I didn't break anything? :smile:

File: tests/test_cmdline.py
Comment (v1kko) [inline tests/test_cmdline.py:1]: What is replacing this file / can I still test the command-line? or is it API only?

**Review by v1kko — APPROVED** [review]
I have added some changes, especially in the get_system() that returned a class, but I think you rewrote it such that it should return an Instance?

 Otherwise it looks good, just one question about the cmdline tests.

File: tests/test_cmdline.py
Comment (v1kko) [reply tests/test_cmdline.py:1]: Doesn't matter actually, merging it now so I can continue with this branch :+1:

**Pull request merged**